contracts/0.4.24/Lido.sol

```diff
@@ -79,33 +97,145 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
-    * @param depositContract official ETH2 Deposit contract
+    * @param _depositContract official ETH2 Deposit contract
     * @param _oracle oracle contract
     * @param _operators instance of Node Operators Registry
+    * @param _treasury contract which accumulates treasury fee
+    * @param _insuranceFund contract which accumulates insurance fee
     */
     function initialize(
-        IDepositContract depositContract,
+        IDepositContract _depositContract,
         address _oracle,
         INodeOperatorsRegistry _operators,
         address _treasury,
         address _insuranceFund
     )
         public onlyInit
     {
-        _setDepositContract(depositContract);
-        _setOracle(_oracle);
-        _setOperators(_operators);
-        _setTreasury(_treasury);
-        _setInsuranceFund(_insuranceFund);
+        NODE_OPERATORS_REGISTRY_POSITION.setStorageAddress(address(_operators));
+        DEPOSIT_CONTRACT_POSITION.setStorageAddress(address(_depositContract));
+
+        _setProtocolContracts(_oracle, _treasury, _insuranceFund);
+
+        STAKE_LIMIT_POSITION.setStorageUint256(
+            StakeLimitUtils.encodeStakeLimitSlot(0, 0, 0, uint32(block.number))
+        );
 
         initialized();
     }
 
+    /**
+    * @notice Cut-off new stake (every new staking transaction submitting user-provided ETH
+    * would revert if `pauseStake` was called previously).
+    * @dev A way to pause stake without pushing PAUSE for the whole proto.
+    * The main goal is to prevent huge APR losses for existing stakers due to high demands
+    * on post-Merge entry queue.
+    * Emits `StakingPaused` event.
+    */
+    function pauseStaking() external {
+        _auth(STAKING_PAUSE_ROLE);
+
+        STAKE_LIMIT_POSITION.setStorageUint256(0);
+        emit StakingPaused();
+    }
+
+    /**
+    * @notice Resume staking if `pauseStaking` was called previously (allow new submits transactions)
+    * or if new rate-limit params are required.
+    *
+    * Staking could be rate-limited by imposing a limit on the stake amount
+    * at each moment in time.
+    *
+    * ▲ Stake limit
+    * │.....  .....   ........ ...            ....     ... Stake limit = max
+    * │      .       .        .   .   .      .    . . .
+    * │     .       .              . .  . . .      . .
+    * │            .                .  . . .
+    * │──────────────────────────────────────────────────> Time
+    * │     ^      ^          ^   ^^^  ^ ^ ^     ^^^ ^     Stake events
+    *
+    * To disable rate-limit pass zero arg values.
+    * @dev Reverts if:
+    * - `_maxStakeLimit` < `_stakeLimitIncreasePerBlock`
+    * - `_maxStakeLimit` / `_stakeLimitIncreasePerBlock` >= 2^32 (only if `_stakeLimitIncreasePerBlock` != 0)
+    * Emits `StakeResumed` event
+    * @param _maxStakeLimit max stake limit value
+    * @param _stakeLimitIncreasePerBlock stake limit increase per single block
+    */
+    function resumeStaking(
+        uint96 _maxStakeLimit,
```

Comment:
it's better to have `uint256` argument type for external functions and explicitly check numeric bounds. this will procude more explicit errors on overflow and cheaper gas-wise

Reply:
(not critical, can leave as is)